wd_comp.c

```diff
@@ -471,20 +470,11 @@ int wd_do_comp_sync2(handle_t h_sess, struct wd_comp_req *req)
 
 	strm_req.last = 0;
 	while (1) {
-		if (req->src_len > chunk) {
-			strm_req.src_len = chunk;
-			req->src_len -= chunk;
-		} else {
-			strm_req.src_len = req->src_len;
-			req->src_len = 0;
-		}
-		avail_in = strm_req.src_len;
-		if (req->op_type == WD_DIR_COMPRESS)
-			strm_req.last = (strm_req.src_len == chunk) ? 0 : 1;
-
+		strm_req.src_len = avail_in > chunk ? chunk : avail_in;
 		do {
+			if (req->op_type == WD_DIR_COMPRESS)
+				strm_req.last = (strm_req.src_len == chunk) ? 0 : 1;
 			if (req->op_type == WD_DIR_COMPRESS &&
-			    strm_req.src_len == 0 &&
 			    strm_req.last == 1) {
```

Comment:
This condition is necessary for the function 'append_store_block'. It only works for the empty block and deals with the 'checksum' or 'isize'.